src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/openstack/OpenstackJCloudsProvider.java

```diff
@@ -17,48 +15,123 @@
 import org.ow2.proactive.connector.iaas.model.Instance;
 import org.springframework.stereotype.Component;
 
-import lombok.Getter;
+import javax.ws.rs.ClientErrorException;
+import javax.ws.rs.NotSupportedException;
+import javax.ws.rs.core.Response;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 @Component
 public class OpenstackJCloudsProvider extends JCloudsProvider {
 
-	@Getter
-	private final String type = "openstack-nova";
+    private final static String region = "RegionOne";
+
+    @Getter
+    private final String type = "openstack-nova";
+
+    @Override
+    public Set<Instance> createInstance(Infrastructure infrastructure, Instance instance) {
+
+        ComputeService computeService = getComputeServiceFromInfastructure(infrastructure);
+
+        NovaApi novaApi = computeService.getContext().unwrapApi((NovaApi.class));
+
+        ServerApi serverApi = novaApi.getServerApi(region);
+
+        String script = buildScriptToExecuteString(instance.getInitScript());
+
+        CreateServerOptions serverOptions = new CreateServerOptions()
+                .keyPairName(instance.getCredentials().getPublicKeyName()).userData(script.getBytes());
+
+        return IntStream.rangeClosed(1, Integer.valueOf(instance.getNumber()))
+                .mapToObj(i -> createOpenstackInstance(instance, serverApi, serverOptions))
+                .map(server -> instanceCreatorFromNodeMetadata.apply(server, infrastructure.getId()))
+                .collect(Collectors.toSet());
+
+    }
+
+    @Override
+    public String addToInstancePublicIp(Infrastructure infrastructure, String instanceId) {
+
+        ComputeService computeService = getComputeServiceFromInfastructure(infrastructure);
+
+        NovaApi novaApi = computeService.getContext().unwrapApi(NovaApi.class);
+        FloatingIPApi api;
+        if (!novaApi.getFloatingIPApi(region).isPresent()) {
+            throw new NotSupportedException("Operation not supported for this Openstack cloud");
+        } else {
```

Comment:
no need to put the else and you can do 

`FloatingIPApi api = novaApi.getFloatingIPApi(region).get();`